src/plot_hourly_line_graph.py

```diff
@@ -3,32 +3,61 @@
 import datetime
 import csv
 import numpy
+import os
+
+directory = r"..\graphs\hourly_line_graph"
+if not os.path.exists(directory):
+    os.makedirs(directory)
 
 list = []
-x = []
-y = []
-bins_x = []
 
 with open('../data/data.csv', 'r') as csvfile:
-    plots = csv.reader((l.replace('\0', '') for l in csvfile))
+    plots = csv.reader(csvfile, delimiter=',')
     for row in plots:
         try:
             list.append([datetime.datetime.strptime(
-                row[0], '%m/%d/%y %H:%M'), int(row[1])])
+                row[0], '%Y-%m-%d %H:%M:%S.%f'), int(row[1])])
         except:
             pass
 
-bins_x = []
-bins_y = []
+bins_x = set()
+date_dict = {}
```

Comment:
It took me a while to understand what this dictionary was used for. This dictionary dates as keys and the values are a list of sensor readings correct?

Maybe something a name like `sensor-reading_day_map` would be more descriptive